src/assets/js/gfpdf-settings/FormSettingsListPage/setupAJAXListDuplicateListener.js

```diff
@@ -0,0 +1,152 @@
+import { ajaxCall } from '../Helper/ajaxCall'
+import { spinner } from '../Helper/Spinner'
+
+/**
+ * Handles the duplicate of a PDF list item via AJAX and fixes up all the nonce actions
+ * @return void
+ * @since 4.0
+ */
+export function setupAJAXListDuplicateListener ($) {
+  /**
+   * Display a message or error to the user with an appropriate timeout
+   * @param  String msg     The message to be displayed
+   * @param  Integer timeout How long to show the message
+   * @param  Boolean error   Whether to show an error (true) or a message (false or undefined)
+   * @return void
+   * @since 4.0
+   */
+  function showMessage (msg, timeout, error) {
+    timeout = typeof timeout !== 'undefined' ? timeout : 4500
+    error = typeof error !== 'undefined' ? error : false
+
+    let $elm = $('<div id="message">').html('<p>' + msg + '</p>')
+
+    if (error === true) {
+      $elm.addClass('error')
+    } else {
+      $elm.addClass('updated')
+    }
+
+    $('.wrap > h2').after($elm)
+
+    setTimeout(function () {
+      $elm.slideUp()
+    }, timeout)
+  }
+
+  /**
+   * Update the URL parameter
+   * @param String    The URL to parse
+   * @param String    The URL parameter to want to update
+   * @param String    The replacement string for the URL parameter
+   * @return String   The processed URL
+   * @since 4.0
+   * @link http://stackoverflow.com/a/10997390/11236
+   */
+  function updateURLParameter (url, param, paramVal) {
```

Comment:
This can be abstracted out.

Reply:
Not necessary I think. Used only in 1 file for its particular function.

Reply:
This is only used in this 1 file for now. The work we're doing is to future-proof the application.